packages/sitecore-jss-angular/src/components/placeholder.component.spec.ts

```diff
@@ -844,3 +896,368 @@ describe('SXA components', () => {
     })
   );
 });
+
+describe('Placeholder Metadata:', () => {
+  const {
+    layoutData,
+    layoutDataWithEmptyPlaceholder,
+    layoutDataWithUnknownComponent,
+  } = metadataData;
+
+  @Component({
+    selector: 'test-text',
+    template: `
+      <div className="nested-test-wrapper"></div>
+    `,
+  })
+  class TestTextComponent {
+    @Input() rendering: ComponentRendering;
+  }
+
+  let fixture: ComponentFixture<TestPlaceholderComponent>;
+  let de: DebugElement;
+  let comp: TestPlaceholderComponent;
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        declarations: [TestTextComponent, TestPlaceholderComponent],
+        imports: [
+          RouterTestingModule,
+          JssModule.withComponents([{ name: 'Home', type: TestTextComponent }]),
+        ],
+        providers: [],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(TestPlaceholderComponent);
+      de = fixture.debugElement;
+
+      comp = fixture.componentInstance;
+      const jssContext = de.injector.get(JssStateService);
+      jssContext.setState(layoutData);
+      fixture.detectChanges();
+    })
+  );
+
+  it(
+    'should render code blocks around nested placeholder components with DEFAULT_PLACEHOLDER_UID',
+    waitForAsync(async () => {
+      const component = layoutData.sitecore.route;
+      const phKey = 'main';
+      comp.name = phKey;
+      comp.rendering = (component as unknown) as ComponentRendering;
+
+      fixture.detectChanges();
+      await fixture.whenStable();
+      fixture.detectChanges();
+
+      expect(de.children.length).toBe(1);
+      const elements = de.queryAll(By.css('code'));
+      expect(elements.length).toBe(4);
+      expect(elements[0].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="open" chrometype="placeholder" id="main_00000000-0000-0000-0000-000000000000"></code>'
+      );
+      expect(elements[1].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="open" chrometype="rendering" id="nested123"></code>'
+      );
+      expect(elements[2].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="close" chrometype="rendering"></code>'
+      );
+      expect(elements[3].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="close" chrometype="placeholder"></code>'
+      );
+    })
+  );
+
+  it(
+    'should render code blocks around nested placeholder components with provided placeholder ID',
```

Comment:
This unit test doesn't cover the case that we expected to cover. It doesn't render nested placeholder and doesn't have "assert" statement for it

Reply:
fixed.